addons/knowledge/static/src/js/knowledge_controller.js

```diff
@@ -39,16 +39,42 @@ const KnowledgeFormController = FormController.extend({
         this.knowledgeFormController = true;
         this.renderer = renderer;
         this._super.apply(this, arguments);
+        this.onFieldSavedListeners = new Map();
+    },
+
+    /**
+     * @override
+     */
+    start: async function () {
+        await this._super.apply(this, arguments);
+        this.onFieldSaved('is_user_favourite', () => {
+            const { id } = this.getState();
+            this._rpc({
+                route: '/knowledge/get_favourite_tree',
+                params: {
+                    res_id: id,
+                }
+            }).then(template => {
+                const $dom = $(template);
+                this.$('.o_favourite_container').replaceWith($dom);
+                this.renderer._setTreeFavoriteListener();
+                this.renderer._renderEmojiPicker($dom);
+            });
+        });
+        this.onFieldSaved('icon', unicode => {
+            const { id } = this.getState();
+            this.renderer._setEmoji(id, unicode);
+        });
     },
 
     // Listeners:
 
-    _onAddRandomIcon: async function() {
-        let id = await this._getId();
```

Comment:
you should not remove that (same for every _getId() call

Reply:
In the `onFieldSaved` callback functions, I do not need to call `_getId` as the callback functions are called right after calling `saveRecord`. The `res_id` should be set properly.

For the `_onAddRandomIcon` function, we do not need the id of the article. We can just propagate a `field_change` event and that's it :)